### DISCUSSION TITLE
[BUG] Get-PnPAzureACSPrincipal Insufficient privileges error when -Scope is Tenant or All

### DISCUSSION THREAD

**Discussion by Ron-Rohlfs** [pnp/powershell#4385] [General]
### Notice
Many bugs reported are actually related to the PnP Framework which is used behind the scenes. Consider carefully where to report an issue:

1. **Are you using ```Invoke-PnPSiteTemplate``` or ```Get-PnPSiteTemplate```**? The issue is most likely related to the Provisioning Engine. The Provisioning engine is _not_ located in the PowerShell repo. Please report the issue here: https://github.com/pnp/pnpframework/issues.
1. **Is the issue related to the cmdlet itself, its parameters, the syntax, or do you suspect it is the code of the cmdlet that is causing the issue?** Then please continue reporting the issue in this repo.
1. **If you think that the functionality might be related to the underlying libraries that the cmdlet is calling** (We realize that might be difficult to determine), please first double check the code of the cmdlet, which can be found here: https://github.com/pnp/powershell/tree/master/src/Commands. If related to the cmdlet, continue reporting the issue here, otherwise report the issue at https://github.com/pnp/pnpframework/issues

### Reporting an Issue or Missing Feature
Get-PnPAzureACSPrincipal returns resuts without expiration dates (ValidUntil) as expected. Attempts to run Get-PnPAzureACSPrincipal -Scope Tenant or Get-PnPAzureACSPrincipal -Scope All returns: 
Get-PnPAzureACSPrincipal: Insufficient privileges to complete the operation.

### Expected behavior 
Expect results that include expiration dates (ValidUntil).

### Actual behavior 
Get-PnPAzureACSPrincipal: Insufficient privileges to complete the operation.

### Steps to reproduce behavior
Connect-PnPOnline -Url "https://xxxxxx-admin.sharepoint.com" -ClientId "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx";
(enter user ans password)
Get-PnPAzureACSPrincipal -Scope Tenant;
Get-PnPAzureACSPrincipal: Insufficient privileges to complete the operation.

### What is the version of the Cmdlet module you are running?
2.12.0

### Which operating system/environment are you running PnP PowerShell on?
- [ X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
Maybe you are missing some permissions on the EntraID app or the account that you are using doesn't have SP admin permissions ?
Works for me.

<img width="764" alt="image" src="https://github.com/user-attachments/assets/b83c9810-dbbc-470d-95b3-54699b010536">

<img width="644" alt="image" src="https://github.com/user-attachments/assets/4dced072-bcfb-46e4-ad36-09d7ffb14667">

**Reply by Ron-Rohlfs** [reply]
account is a GA has SP admin perissions
